superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx

```diff
@@ -65,8 +66,13 @@ export const DndColumnSelect = (props: LabelProps) => {
     ) {
       onChange(optionSelectorValues);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [JSON.stringify(value), JSON.stringify(optionSelector.getValues())]);
+    // when options change, that means that the dataset has changed
+    // so we have to check if values are still applicable.
+  }, [options, value, optionSelector]);
```

Comment:
If options change, we'll get a new instance of optionSelector, so I don't think we need to include it in dependency array here

Reply:
That's true but I felt like appeasing the linter